crates/datasources/src/bson/mod.rs

```diff
@@ -0,0 +1,281 @@
+use datafusion::arrow::array::cast::as_string_array;
+use datafusion::arrow::array::{types::*, Array, ArrayRef, AsArray, StructArray};
+use datafusion::arrow::datatypes::{DataType, Fields, IntervalUnit, TimeUnit};
+use datafusion::arrow::error::ArrowError;
+
+use bson;
+
+pub struct BsonBatchConverter {
+    batch: StructArray,
+    schema: Vec<String>,
+    row: usize,
+    started: bool,
+    columns: Vec<Vec<bson::Bson>>,
+}
+
+impl BsonBatchConverter {
+    pub fn new(batch: StructArray, fields: Fields) -> Self {
+        let mut field_names = Vec::<String>::with_capacity(fields.len());
+        for field in &fields {
+            field_names.push(field.name().to_owned())
+        }
+
+        Self {
+            batch: batch.clone(),
+            schema: field_names,
+            row: 0,
+            started: false,
+            columns: Vec::<Vec<bson::Bson>>::with_capacity(batch.num_columns()),
+        }
+    }
```

Comment:
```suggestion
    pub fn new(batch: StructArray, fields: Fields) -> Self {
        let mut field_names = Vec::with_capacity(fields.len());
        for field in &fields {
            field_names.push(field.name().to_owned())
        }

        Self {
            batch: batch.clone(),
            schema: field_names,
            row: 0,
            started: false,
            columns: Vec::with_capacity(batch.num_columns()),
        }
    }
```

Types can be inferred here.

Reply:
I don't like it 😂

Reply:
I'm fine with whatever here and just wanted to point out it's an option.